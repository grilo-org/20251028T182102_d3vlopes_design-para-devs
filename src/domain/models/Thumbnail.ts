import { ImageModel } from './Image';

export interface ThumbnailModel {
	image: ImageModel;
}
