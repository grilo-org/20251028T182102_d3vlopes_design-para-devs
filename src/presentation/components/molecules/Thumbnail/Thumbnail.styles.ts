import { styleWrapper } from '@/libs/styles/tailwind-variants';

export const styles = styleWrapper({
	slots: {
		wrapperStyles:
			'relative flex justify-center items-center h-[14.0625rem] md:h-[24.0413rem]',
		imageStyles: 'rounded-[4px] w-full md:h-[24.0413rem]',
		imageOverlayStyles: 'absolute inset-0 bg-black opacity-80 rounded-[4px]',
		buttonStyles: 'absolute m-auto',
		iconStyles: 'w-12 h-12 md:h-[11.5875rem] md:w-[11.5875rem]',
	},
});

const makeStyles = styles();

export const classNames = {
	wrapperStyles: makeStyles.wrapperStyles(),
	imageStyles: makeStyles.imageStyles(),
	imageOverlayStyles: makeStyles.imageOverlayStyles(),
	buttonStyles: makeStyles.buttonStyles(),
	iconStyles: makeStyles.iconStyles(),
};
