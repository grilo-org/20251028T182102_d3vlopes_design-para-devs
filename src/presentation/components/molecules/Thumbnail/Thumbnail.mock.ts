import { ThumbnailModel } from '@/domain/models/Thumbnail';

export const thumbnailMock: ThumbnailModel = {
	image: {
		src: 'https://media.graphassets.com/I7TeV9zVSBiR8bkitszw',
		alt: 'Capa do eBook, escrito Design para Devs com um programador em frente ao computador',
	},
};
