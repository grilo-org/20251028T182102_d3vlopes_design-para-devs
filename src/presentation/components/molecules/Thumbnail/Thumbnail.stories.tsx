import type { Meta, StoryObj } from '@storybook/react';

import { thumbnailMock } from './Thumbnail.mock';

import { Thumbnail } from '.';

const meta: Meta<typeof Thumbnail> = {
	title: 'Molecules/Thumbnail',
	component: Thumbnail,
	args: {
		...thumbnailMock,
	},
	argTypes: {
		onClickPlayButton: { action: 'clicked' },
	},
};

export default meta;
type Story = StoryObj<typeof Thumbnail>;

const template: Story = {
	render: (args) => (
		<div className="max-w-[1120px] m-auto">
			<Thumbnail {...args} />
		</div>
	),
};

export const Default: Story = {
	...template,
};
