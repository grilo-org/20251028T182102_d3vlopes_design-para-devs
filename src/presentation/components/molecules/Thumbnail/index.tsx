import { ThumbnailModel } from '@/domain/models/Thumbnail';

import { Image } from '@/presentation/components/atoms/Image';

import { PlayIcon } from './assets/PlayIcon';

import { classNames } from './Thumbnail.styles';

export interface ThumbnailProps extends ThumbnailModel {
	onClickPlayButton: () => void;
}

export const Thumbnail = ({ image, onClickPlayButton }: ThumbnailProps) => {
	return (
		<div className={classNames.wrapperStyles}>
			<Image
				className={classNames.imageStyles}
				fill
				src={image.src}
				alt={image.alt}
			/>

			<div className={classNames.imageOverlayStyles} />

			<button
				type="button"
				className={classNames.buttonStyles}
				onClick={onClickPlayButton}
			>
				<PlayIcon />
			</button>
		</div>
	);
};
