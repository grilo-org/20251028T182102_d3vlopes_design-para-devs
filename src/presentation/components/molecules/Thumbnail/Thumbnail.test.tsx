import { fireEvent, render, screen } from '@testing-library/react';

import { thumbnailMock } from './Thumbnail.mock';

import { Thumbnail, ThumbnailProps } from '.';

const { getByRole, getByLabelText } = screen;

const mock: ThumbnailProps = {
	...thumbnailMock,
	onClickPlayButton: jest.fn(),
};

describe('<Thumbnail />', () => {
	beforeEach(() => {
		render(<Thumbnail {...mock} />);
	});

	it('should render image', () => {
		const image = getByRole('img', { name: mock.image.alt });

		expect(image).toBeInTheDocument();
	});

	it('should render button', () => {
		const button = getByRole('button');

		expect(button).toBeInTheDocument();
	});

	it('should call function on click button', () => {
		const button = getByRole('button');

		fireEvent.click(button);

		expect(mock.onClickPlayButton).toHaveBeenCalled();
	});

	it('should render icon', () => {
		const icon = getByLabelText('Ícone representando o inicio da reprodução');

		expect(icon).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const { container } = render(<Thumbnail {...mock} />);

		expect(container).toMatchSnapshot();
	});
});
