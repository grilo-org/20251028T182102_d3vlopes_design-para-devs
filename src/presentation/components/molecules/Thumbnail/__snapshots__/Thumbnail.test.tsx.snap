// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thumbnail /> should match snapshot 1`] = `
<div>
  <div
    class="relative flex justify-center items-center h-[14.0625rem] md:h-[24.0413rem]"
  >
    <img
      alt="Capa do eBook, escrito Design para Devs com um programador em frente ao computador"
      class="rounded-[4px] w-full md:h-[24.0413rem]"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FI7TeV9zVSBiR8bkitszw&w=3840&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FI7TeV9zVSBiR8bkitszw&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FI7TeV9zVSBiR8bkitszw&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FI7TeV9zVSBiR8bkitszw&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FI7TeV9zVSBiR8bkitszw&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FI7TeV9zVSBiR8bkitszw&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FI7TeV9zVSBiR8bkitszw&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FI7TeV9zVSBiR8bkitszw&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FI7TeV9zVSBiR8bkitszw&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
    <div
      class="absolute inset-0 bg-black opacity-80 rounded-[4px]"
    />
    <button
      class="absolute m-auto"
      type="button"
    >
      <svg
        aria-label="Ícone representando o inicio da reprodução"
        class="w-12 h-12 md:h-[11.5875rem] md:w-[11.5875rem]"
        fill="none"
        viewBox="0 0 186 186"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Ícone representando o inicio da reprodução
        </title>
        <circle
          cx="93.2699"
          cy="92.7786"
          r="89.6986"
          stroke="#8133F1"
          stroke-width="6"
        />
        <path
          d="M59.3399 34.01L161.13 92.7786L59.3399 151.547L59.3399 34.01Z"
          fill="#9654F4"
        />
      </svg>
    </button>
  </div>
</div>
`;
